File: smarttrade_app/lib/features/chart_analysis/domain/repositories/chart_annotation_repository.dart
import 'package:dartz/dartz.dart';
import 'package:smarttrade_app/core/error/failures.dart';
import 'package:smarttrade_app/features/chart_analysis/domain/entities/chart_annotation.dart';

abstract class ChartAnnotationRepository {
  Future<Either<Failure, List<ChartAnnotation>>> getAnnotations(
    String instrumentSymbol,
  );

  Future<Either<Failure, Unit>> saveAnnotation(ChartAnnotation annotation);

  Future<Either<Failure, Unit>> deleteAnnotation(String id);
}

File: smarttrade-web/src/features/chart-analysis/domain/repositories/chartAnnotationRepository.ts
import type { Result } from '@/core/result/result'
import type { ChartAnnotation } from '../entities/chartAnnotation'

export interface ChartAnnotationRepository {
  getAnnotations(instrumentSymbol: string): Promise<Result<ChartAnnotation[]>>
  saveAnnotation(annotation: ChartAnnotation): Promise<Result<void>>
  deleteAnnotation(id: string): Promise<Result<void>>
}
